### PR TITLE
Implemented #1071: Invoke NFC in Advanced View

### PR DESCRIPTION
**Pull request by kentnek**
For Issue #1071:

 - Extracted NFC code from `ViewKeyActivity` to `NfcHelper` to share code between classes
 - Changed the private anonymous `Handler` for NFC into a static private subclass, that uses `WeakReference` to avoid memory leaks
 - Added resources needed (retrieved from Graphics) for the NFC button inside `ViewKeyAdvShareFragment`.
 - Fixed the ripple boundary of the `Share With..`. button to prevent it from bleeding to other buttons on the right (UX improvement)

image

However I'm wondering if both key types (owned and not owned) can be shared via NFC, why not just keep the `Share via NFC` button in the ActionBar of `ViewKeyActivity`. Currently its `Visibility` is set to `GONE` if a key doesn't belong to the user.

### CODE REVIEW

File: OpenKeychain/src/main/java/org/sufficientlysecure/keychain/util/NfcHelper.java [hunk 1]
```diff
@@ -0,0 +1,204 @@
+package org.sufficientlysecure.keychain.util;
```
Comment (dschuermann) [inline OpenKeychain/src/main/java/org/sufficientlysecure/keychain/util/NfcHelper.java]: Please add the GPLv3 header here like in the other classes

File: OpenKeychain/src/main/java/org/sufficientlysecure/keychain/util/NfcHelper.java [hunk 2]
```diff
@@ -0,0 +1,204 @@
...
+import org.sufficientlysecure.keychain.provider.KeychainContract;
+import org.sufficientlysecure.keychain.provider.ProviderHelper;
+import org.sufficientlysecure.keychain.ui.util.Notify;
+
+import java.lang.ref.WeakReference;
+
+/**
+ * This class contains NFC functionality that can be shared across Fragments or Activities.
+ * <p/>
+ * Created on Mar 20, 2015.
```
Comment (dschuermann) [inline OpenKeychain/src/main/java/org/sufficientlysecure/keychain/util/NfcHelper.java]: Remove "created" and author here

**Comment by kentnek** [comment]
Oh okay

**Comment by kentnek** [comment]
I'm not really sure about the GNU header though, so may I put in my name inside as well like the one I have just committed?

**Comment by dschuermann** [comment]
Yes exactly, the GNU header with your name is enough.

**Comment by kentnek** [comment]
Ok thanks a lot!

**Comment by dschuermann** [comment]
Regarding:
> However I'm wondering if both key types (owned and not owned) can be shared via NFC, why not just keep the Share via NFC button in the ActionBar of ViewKeyActivity. Currently its Visibility is set to GONE if a key doesn't belong to the user.

We don't want to encourage the sharing of keys that do not belong to you, but we want to allow it via the advanced view for advanced users. Average users should only share their own keys.

**Comment by kentnek** [comment]
Oh i see.

**Comment by dschuermann** [comment]
Forgot to add: Thanks for your work ;)

**Comment by kentnek** [comment]
It's my pleasure :) I also need to complete the one-patch requirement for GSOC xD

**Pull request merged**